### ISSUE TITLE
Selenium standalone server not pushed to maven

### ISSUE THREAD

**Issue by Mobrockers** [SeleniumHQ/selenium#1928]
## Meta -
OS: any
Selenium Version: grid
Browser: any
Browser Version: any

I use the jenkins selenium grid plugin found here: https://wiki.jenkins-ci.org/display/JENKINS/Selenium+Plugin
This plugin uses selenium standalone server and downloads this from maven. Selenium team has decided some time (years?) ago to no longer provide the selenium standalone server jar on maven. As a workaround the plugin maintainer implemented an ant task that downloads the jar from the selenium hosting site, then uploads it to the jenkins private maven repository. This however requires that a jenkins core team member runs a script every time a selenium update comes out. The plugin maintainer himself is not a jenkins core team member. Due to this the plugin is stuck on selenium version 2.41.

I would like selenium to once again provide the selenium server standalone jar through maven, so that the jenkins plugin can use a more recent version more easily. Is this possible?

The jenkins selenium plugin currently has more than 1700 installs. I do not believe this is an insignificant part of the selenium user base, is it?

**Comment by lukeis** [comment]
We have never pushed the standalone jar to maven, if one uses maven the server module should suffice. Otherwise the plugin should be modified to download from our Google storage location (which can dynamically figure out the latest by parsing the xml returned on the base url)
https://selenium-release.storage.googleapis.com

**Comment by Mobrockers** [comment]
@lukeis The plugin does download from the Google storage location, then it uploads the jar to maven. There is little documentation on the plugin unfortunately, and the maintainer has been absent for a while, so I have not been able to figure out why this convoluted workaround was chosen.
I have tried to change the dependency to use the server module, but the server would not start when I did this because the plugin currently calls the standalone jar using 
`[Jenkins] $ "C:\Program Files (x86)\Jenkins\jre\bin\java" -cp "C:\Program Files (x86)\Jenkins\war\WEB-INF\lib\remoting-2.53.2.jar" hudson.remoting.Launcher -cp "C:\Program Files (x86)\Jenkins\plugins\selenium\WEB-INF\lib\selenium-server-standalone-2.41.0.jar" -connectTo localhost:59526`

The selenium standalone server jar is also packaged with the plugin, instead of downloaded at runtime. I do not think this was the best decision, but the codebase is too complicated for me to change this right now, so I thought it would be easier if the standalone jar would simply be available in a publicly accessible maven repository.


I fully agree with you that the plugin needs major changes in this regard, but I do not have the time or skill at the moment to implement this, and no one seems to be interested in supporting the plugin at the moment, even though it is a very valuable plugin.
